### ISSUE TITLE
[Feature Request] Pass ImageCreateFlags into ImmutableImage::from_buffer and from_iter

### ISSUE THREAD

**Issue by gregjohnson2017** [vulkano-rs/vulkano#1988]
I ran across an issue while trying to create an ImmutableImage to back a CubeArray. I'm getting an error saying ImageNotCubeCompatible, which is explained in the [documentation](https://docs.rs/vulkano/latest/vulkano/image/view/enum.ImageViewCreationError.html#variant.ImageNotCubeCompatible):

> ImageNotCubeCompatible
> A cube image view type was requested, but the image was not created with the cube_compatible flag.

I see that `ImmutableImage::from_iter` is calling `ImmutableImage::from_buffer`, which has the flags [hard coded to be empty](https://github.com/vulkano-rs/vulkano/blob/8f99ceab3257d0a12a3c39ebe8206364c0229c7c/vulkano/src/image/immutable.rs#L278):
```rust
let flags = ImageCreateFlags::empty();
```
I believe that ImmutableImage is the appropriate type to use for CubeArray, because I just intend to write once and never again. It looks like as a workaround I could use `ImmutableImage::uninitialized` to specify the flags, but then I would have replicate the functionality of `ImmutableImage::from_buffer` to create a command buffer builder and perform the one-time write into the initializer, and also potentially generate mipmaps.

In summary, my request is that `ImageCreateFlags` should be added as a parameter to `ImmutableImage::from_iter` and `ImmutableImage::from_buffer`.

**Comment by gregjohnson2017** [comment]
We got our cube map array working with a StorageImage. We had lots of problems trying to get the cube map working with an ImmutableImage (with the workarounds), so it seems like we might be wrong about that being an appropriate method of storage. I'm not sure if this feature request is valid, since it's clear that there's a lot I don't understand about this. I would appreciate if someone could explain why I'm wrong about thinking ImmutableImage is right for a cube map array.

**Comment by comrademarc** [comment]
You are right, there are many issues with the way images are currently and to a lesser extent buffers. Lots of things are hard-coded. Another issue I encountered is that you can't create a storage image with mip-maps, for example. The plan is that there are going to be the single types `Image` and `Buffer` in the future. It is pretty high up my todo-list, so I'll be working on that soon. Until then, please use some workaround, because you would need to fix things once we unite the types either way.
